python_ta/reporters/node_printers.py

```diff
@@ -134,64 +134,16 @@ def render_missing_space_in_doctest(msg, _node, source_lines=None):
-    elif "E116" in msg.msg:
-        yield from render_pep8_errors_e116(msg, _node, source_lines)
-    elif "E122" in msg.msg or "E127" in msg.msg or "E131" in msg.msg:
-        yield from render_pep8_errors_e122_and_e127_and_e131(msg, _node, source_lines)
-    elif "E124" in msg.msg:
-        yield from render_pep8_errors_e124(msg, _node, source_lines)
-    elif "E125" in msg.msg or "E129" in msg.msg:
-        yield from render_pep8_errors_e125_and_e129(msg, _node, source_lines)
-    elif "E128" in msg.msg:
-        yield from render_pep8_errors_e128(msg, _node, source_lines)
-    elif "E201" in msg.msg or "E202" in msg.msg or "E203" in msg.msg:
-        yield from render_pep8_errors_e201_e202_e203_e211(msg, _node, source_lines)
-    elif "E221" in msg.msg:
-        yield from render_pep8_errors_e221(msg, _node, source_lines)
-    elif "E222" in msg.msg:
-        yield from render_pep8_errors_e222(msg, _node, source_lines)
-    elif "E223" in msg.msg:
-        yield from render_pep8_errors_e223(msg, _node, source_lines)
-    elif "E224" in msg.msg or "E273" in msg.msg:
-        yield from render_pep8_errors_e224_and_e273(msg, _node, source_lines)
-    elif "E226" in msg.msg:
-        yield from render_pep8_errors_e226(msg, _node, source_lines)
-    elif "E227" in msg.msg:
-        yield from render_pep8_errors_e227(msg, _node, source_lines)
-    elif "E228" in msg.msg:
-        yield from render_pep8_errors_e228(msg, _node, source_lines)
-    elif "E251" in msg.msg:
-        yield from render_pep8_errors_e251(msg, _node, source_lines)
-    elif "E261" in msg.msg:
-        yield from render_pep8_errors_e261(msg, _node, source_lines)
-    elif "E262" in msg.msg:
-        yield from render_pep8_errors_e262(msg, _node, source_lines)
-    elif "E265" in msg.msg:
-        yield from render_pep8_errors_e265(msg, _node, source_lines)
-    elif "E266" in msg.msg:
-        yield from render_pep8_errors_e266(msg, _node, source_lines)
-    elif "E272" in msg.msg:
-        yield from render_pep8_errors_e272(msg, _node, source_lines)
-    elif "E275" in msg.msg:
-        yield from render_pep8_errors_e275(msg, _node, source_lines)
-    elif "E301" in msg.msg:
-        yield from render_pep8_errors_e301(msg, _node, source_lines)
-    elif "E302" in msg.msg:
-        yield from render_pep8_errors_e302(msg, _node, source_lines)
-    elif "E303" in msg.msg:
-        yield from render_pep8_errors_e303(msg, _node, source_lines)
-    elif "E304" in msg.msg:
-        yield from render_pep8_errors_e304(msg, _node, source_lines)
-    elif "E305" in msg.msg:
-        yield from render_pep8_errors_e305(msg, _node, source_lines)
-    elif "E306" in msg.msg:
-        yield from render_pep8_errors_e306(msg, _node, source_lines)
-    else:
-        yield from render_generic(msg, _node, source_lines)
+
+    # Check if any error codes from the ERROR_TO_ERROR_FUNCTION_MAPPING dict are present in the message
+    for error in ERROR_TO_ERROR_FUNCTION_MAPPING:
```

Comment:
This is certainly a big improvement over the original, but can be made more efficient. `msg.msg` will have a standard format, and so it should be possible to use a regex to extract the error code from the string, rather than looping over the dictionary.